src/java/com/puppetlabs/http/client/impl/ClientMetricFilter.java

```diff
@@ -0,0 +1,64 @@
+package com.puppetlabs.http.client.impl;
+
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.Metric;
+import org.apache.commons.lang3.StringUtils;
+
+public class ClientMetricFilter {
```

Comment:
Mostly implemented this way because I don't know how to java, and initially when implementing them I had thought that there would end up being even more if we ended up adding the `response-init` timers, and that they would all be only a couple lines, and it seemed silly to have 8 files with two lines. However, I think with the way the implementation worked out that it makes more sense to have these all be standalone classes (especially if that's the more java-y way of doing it).